### PR TITLE
fix unrelocatable cmake config

### LINKED ISSUE #421: Problems with caching
There is a general problem, which I'll try to illustrate with an example.

After recent altering of CMAKE_ARGS for 'opusfile' package https://github.com/cpp-pm/hunter/pull/415/commits/d061f8d740ae034e1366b73722c0bca14bd7a8fc

I have a problem with testing of 'SDL_mixer' package, which is dependent on 'opusfile' one
Tests are failing for all systems for which cache had been uploaded https://github.com/drodin/hunter/actions/runs/992722405
Sample of error: https://github.com/drodin/hunter/runs/2969916903?check_suite_focus=true#step:7:8164
```
-- [hunter *** DEBUG *** 2021-07-02T07:01:08] Package installed from cache: opusfile
...
  Imported target "opusfile::opusfile" includes non-existent path
```

Problem goes away if I disable cache server by adding *set(HUNTER_USE_CACHE_SERVERS NO)* to *examples/SDL_mixer/CMakeLists.txt* https://github.com/drodin/hunter/commit/1ad354345088578ade43adc4ebdd9e74a0a08045
Testing goes successful, as 'opusfile' package is built from source
https://github.com/drodin/hunter/actions/runs/992745793

### PR DESCRIPTION
**Pull request by drodin**
fix https://github.com/cpp-pm/hunter/issues/421#issuecomment-920016397

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -20,12 +20,13 @@ if(HUNTER_ENABLED)
...
     set(OGG_LIBRARY libogg::ogg)
     get_target_property(OGG_INCLUDE_DIR libogg::ogg INTERFACE_INCLUDE_DIRECTORIES)
+    set(OGG_INCLUDE_DIR $<BUILD_INTERFACE:${OGG_INCLUDE_DIR}>)
 
     hunter_add_package(Opus)
     find_package(Opus CONFIG REQUIRED)
     set(OPUS_LIBRARY Opus::opus)
     get_target_property(OPUS_INCLUDE_DIR Opus::opus INTERFACE_INCLUDE_DIRECTORIES)
-    set(OPUS_INCLUDE_DIR ${OPUS_INCLUDE_DIR} ${OPUS_INCLUDE_DIR}/opus)
+    set(OPUS_INCLUDE_DIR $<BUILD_INTERFACE:${OPUS_INCLUDE_DIR}> $<BUILD_INTERFACE:${OPUS_INCLUDE_DIR}/opus> $<INSTALL_INTERFACE:include/opus>)
```
Comment (NeroBurner) [inline CMakeLists.txt:29]: for the installed package the include directory needs to be just `include/opus`? So for example when using the package the file in `installdir/include/opus/opus.h` one needs to do `#include <opus.h>` and can't do `#include <opus/opus.h>`? Is this intentional?

**Review by NeroBurner — COMMENTED** [review]
good fix. Only one question as comment and a plea:

Please open a PR upstream as well. Such changes would be good for all users. And we have a smaller patchset with just Hunter specific changes

File: CMakeLists.txt (hunk 1 above)
Comment (drodin) [reply CMakeLists.txt:29]: Hi, this is an old pr from Oct 2021. Upstream has their own CMakeList.txt now. Will look into it after the next release.
One can still do `#include <opus/opus.h>` as "global" _include_ dir is added to **INTERFACE_INCLUDE_DIRECTORIES** by `install (TARGETS ... INCLUDES DESTINATION ${CMAKE_INSTALL_INCLUDEDIR}`) https://github.com/cpp-pm/opusfile/blob/fc28a7d7ffdeaa03d7a33b707dfeb2bce5269130/CMakeLists.txt#L95-L102

**Review by NeroBurner — APPROVED** [review]

**Comment by NeroBurner** [comment]
@rbsheth could you give me the authorization to merge this PR and create a release tag

Screenshot_20220610_125852

**Comment by drodin** [comment]
@rbsheth Hi, found this one also pending, please merge. Thanks.

**Comment by rbsheth** [comment]
Released: https://github.com/cpp-pm/opusfile/releases/tag/v0.12-p2

**Pull request merged**